staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -258,3 +297,158 @@ func (c *CacheDelegator) ReadinessCheck() error {
 func (c *CacheDelegator) RequestWatchProgress(ctx context.Context) error {
 	return c.storage.RequestWatchProgress(ctx)
 }
+
+func (c *CacheDelegator) Stop() {
+	c.stopOnce.Do(func() {
+		close(c.stopCh)
+	})
+	c.wg.Wait()
+}
+
+func newConsistencyChecker(resourcePrefix string, newListFunc func() runtime.Object, cacher getListerReady, etcd getLister, clock clock.Clock) *consistencyChecker {
+	return &consistencyChecker{
+		resourcePrefix: resourcePrefix,
+		newListFunc:    newListFunc,
+		cacher:         cacher,
+		etcd:           etcd,
+		clock:          clock,
+	}
+}
+
+type consistencyChecker struct {
+	resourcePrefix string
+	newListFunc    func() runtime.Object
+
+	cacher getListerReady
+	etcd   getLister
+	clock  clock.Clock
+}
+
+type getListerReady interface {
+	getLister
+	Ready() bool
+}
+
+type getLister interface {
+	GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error
+}
+
+func (c consistencyChecker) startChecking(stopCh <-chan struct{}) {
+	err := wait.PollUntilContextCancel(wait.ContextForChannel(stopCh), ConsistencyCheckPeriod, false, func(ctx context.Context) (done bool, err error) {
+		c.check(ctx)
+		return false, nil
+	})
+	if err != nil {
+		klog.ErrorS(err, "Failed to start consistency checking")
```

Comment:
actually - the only case this can error now is when the `PollUntilContextCancel` timesout/finishes, which we shouldn't even error.

So technically, it's not even about starting...

Reply:
I agree this case should never happen, however static analysis requires me to handle potential error. I can change the message to  `klog.ErrorS(err, "Consistency checking loop exited")`

Reply:
My point is - it will happen in every single run when the component will be stopping.

I know we need to handle an error, but I believe this log will be present always when apiserver is stopping, which will be super misleading...